python/ray/tune/suggest/optuna.py

```diff
@@ -261,33 +293,56 @@ def __init__(self,
                 "You passed an initialized sampler to `OptunaSearch`. The "
                 "`seed` parameter has to be passed to the sampler directly "
                 "and will be ignored.")
+        elif sampler:
+            assert isinstance(
+                self._sampler,
+                BaseSampler), ("You can only pass an instance of "
+                               "`optuna.samplers.BaseSampler` "
+                               "as a sampler to `OptunaSearcher`.")
 
-        self._sampler = sampler or ot.samplers.TPESampler(seed=seed)
-
-        assert isinstance(self._sampler, BaseSampler), \
-            "You can only pass an instance of `optuna.samplers.BaseSampler` " \
-            "as a sampler to `OptunaSearcher`."
+        self._sampler = sampler
+        self._seed = seed
 
         self._ot_trials = {}
         self._ot_study = None
         if self._space:
             self._setup_study(mode)
 
-    def _setup_study(self, mode: str):
+    def _setup_study(self, mode: Union[str, list]):
         if self._metric is None and self._mode:
+            if isinstance(self._mode, list):
+                raise ValueError(
+                    "If ``mode`` is a list (multi-objective optimization "
+                    "case), ``metric`` must be defined.")
             # If only a mode was passed, use anonymous metric
             self._metric = DEFAULT_METRIC
 
         pruner = ot.pruners.NopPruner()
         storage = ot.storages.InMemoryStorage()
 
+        if self._sampler:
+            sampler = self._sampler
```

Comment:
[note] Following samplers don't support multi-objective optimization. So, when `self._mode` is a list and `self._sampler` is one of them, Optuna would raise an exception (You can choose other samplers listed in https://optuna.readthedocs.io/en/stable/reference/samplers.html).

- https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.CmaEsSampler.html#optuna.samplers.CmaEsSampler
- https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.PyCmaSampler.html#optuna.integration.PyCmaSampler
- https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.SkoptSampler.html#optuna.integration.SkoptSampler

Reply:
I think it's fine to let Optuna raise it here. 

BTW. would it be possible for Optuna to add a property to its samplers that would inform whether the sampler can support multi-objective optimization?